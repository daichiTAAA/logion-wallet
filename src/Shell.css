.ShellHeader {
    margin-bottom: 50px;
}
