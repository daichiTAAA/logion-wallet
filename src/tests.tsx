import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
