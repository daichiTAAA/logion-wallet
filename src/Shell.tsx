import React from 'react';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Navbar from 'react-bootstrap/Navbar';

import Logo from './Logo';

import './Shell.css';

export interface Props {
    children: JSX.Element | (JSX.Element | null)[] | null,
}

export default function Shell(props: Props) {
    return (
        <Container>
            <div className="ShellHeader">
                <Row>
                    <Col>
                        <Navbar>
                            <Navbar.Brand>
                                <Logo size={50} />
                            </Navbar.Brand>
                        </Navbar>
                    </Col>
                </Row>
            </div>
            <div className="ShellContent">
                <Row>
                    <Col>
                        {props.children}
                    </Col>
                </Row>
            </div>
        </Container>
    );
}
