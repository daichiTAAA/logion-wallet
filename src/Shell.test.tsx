import React from 'react';
import Shell from './Shell';
import {shallowRender} from './tests';

test('renders', () => {
    const tree = shallowRender(
        <Shell>
            <p>Some text</p>
            <a href="https://logion.network/">A link</a>
        </Shell>
    );
    expect(tree).toMatchSnapshot();
});

test('renders with custom background', () => {
    const tree = shallowRender(
        <Shell backgroundCss="linear-gradient(to right, #a158ff, 90%, #203acf)">
            <p>Some text</p>
            <a href="https://logion.network/">A link</a>
        </Shell>
    );
    expect(tree).toMatchSnapshot();
});
