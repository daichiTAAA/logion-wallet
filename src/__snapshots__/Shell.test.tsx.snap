// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="Shell"
>
  <Container
    fluid={false}
  >
    <div
      className="ShellHeader"
    >
      <Row>
        <Col>
          <Navbar
            collapseOnSelect={false}
            expand={true}
            variant="light"
          >
            <NavbarBrand>
              <Logo
                size={50}
              />
            </NavbarBrand>
          </Navbar>
        </Col>
      </Row>
    </div>
    <div
      className="ShellContent"
    >
      <Row>
        <Col>
          <p>
            Some text
          </p>
          <a
            href="https://logion.network/"
          >
            A link
          </a>
        </Col>
      </Row>
    </div>
  </Container>
</div>
`;

exports[`renders with custom background 1`] = `
<div
  className="Shell"
  style={
    Object {
      "background": "linear-gradient(to right, #a158ff, 90%, #203acf)",
    }
  }
>
  <Container
    fluid={false}
  >
    <div
      className="ShellHeader"
    >
      <Row>
        <Col>
          <Navbar
            collapseOnSelect={false}
            expand={true}
            variant="light"
          >
            <NavbarBrand>
              <Logo
                size={50}
              />
            </NavbarBrand>
          </Navbar>
        </Col>
      </Row>
    </div>
    <div
      className="ShellContent"
    >
      <Row>
        <Col>
          <p>
            Some text
          </p>
          <a
            href="https://logion.network/"
          >
            A link
          </a>
        </Col>
      </Row>
    </div>
  </Container>
</div>
`;
